unit_tests/farm_pipeline_composition.cpp

```diff
@@ -50,8 +50,8 @@ class farm_pipeline_test : public ::testing::Test {
   vector<int> w{};
 
   // entry counter
-  int idx_in = 0;
-  int idx_out = 0;
+  size_t idx_in = 0;
```

Comment:
Why?

Reply:
We have comparision with a .size() in lines 170, 216, 259 and 303.
Since it is a unit test, and indexes should always be positive i think is best to change them to size_t instead of using a static cast every time we compare them

Reply:
I'm not 100% convinced.

But OK. Let's do it for now.

Reply:
Size comparison in line 170. Similar issue pointed out previously. I think the size_t solution is best than a static_cast since it is just a unit test and this code is under our control and not something the user will be "using"